src/btree/bt_split.c

```diff
@@ -986,6 +985,18 @@ __split_parent(WT_SESSION_IMPL *session, WT_REF *ref, WT_REF **ref_new,
 					    session, 0, ikey, size));
 					WT_MEMSIZE_ADD(parent_decr, size);
 				}
+				/*
+				 * The page_del structure can be freed
+				 * immediately: it is only read when the ref
+				 * state is WT_REF_DELETED.  The size of the
+				 * structures wasn't added to the parent: don't
+				 * decrement.
+				 */
+				if (next_ref->page_del != NULL) {
+					__wt_free(session,
+					    next_ref->page_del->update_list);
+					__wt_free(session, next_ref->page_del);
+				}
```

Comment:
We do restore the state, at line 1067:

```c
err:    if (!complete)
                for (i = 0; i < parent_entries; ++i) {
                        next_ref = pindex->index[i];
                        if (next_ref->state == WT_REF_SPLIT)
                                next_ref->state = WT_REF_DELETED;
                }
```

Given that, I think we should free `page_del` at the same time we schedule other memory on the page to be freed (that is, after the split is completed and we're never going to restore the ref states).